db/fmp_to_postgres.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import os
+import pickle
+import sys
+
+import pandas as pd
+import numpy as np
+from tqdm import tqdm
+
+from fmp import FMP
+from psql import PSQL
+from utils import edit_distance
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-d", "--dsn", help="dsn name")
+    parser.add_argument("-u", "--username", help="username for dsn")
+    parser.add_argument("-p", "--password", help="password for dsn")
+    return parser.parse_args()
+
+
+def get_pk_and_fk_from_table(fmp: FMP, df: pd.DataFrame, table_name: str):
+    pk = None
+    fks = []
+    smallest_pk_edit_distance = float("inf")
+    for field_name in df.columns:
+        # extract primary keys
+        if field_name[:3] == "pk_" and field_name[-3:] == "_id":
+            # possible primary key
+            distance = edit_distance(field_name[3:-3], table_name)
+            if distance < smallest_pk_edit_distance:
+                pk = field_name
+                smallest_pk_edit_distance = distance
+
+        # extract foreign keys
+        if field_name[:3] == "fk_" and field_name[-3:] == "_id":
+            fks.append((field_name, fmp.get_tablename_from_fieldname(field_name)))
+
+    return pk, fks
+
+
+def save_keys(filename: str, all_data: dict):
+    np.save(filename, all_data)
+
+
+def load_dict(filename: str):
+    np_load_old = np.load
+    np.load = lambda *a, **k,: np_load_old(*a, allow_pickle=True, **k)
+    np_data = np.load(filename)[()]
+
+    # Restore np.load
+    np.load = np_load_old
+
+    return np_data
+
+def main(argv):
+    fmp = FMP(argv.dsn, argv.username, argv.password)
+    # TODO: extract username, password, and database name from the database.env file
```

Comment:
Have these TODOs been completed?

Reply:
no, they have not, but the ticket has been created #97